nixos/modules/services/databases/openldap.nix

```diff
@@ -244,57 +356,156 @@ in
   };
 
   meta = {
-    maintainers = [ lib.maintainers.mic92 ];
+    maintainers = with lib.maintainters; [ mic92 kwohlfahrt ];
   };
 
-
-  ###### implementation
-
   config = mkIf cfg.enable {
-    assertions = [
-      {
-        assertion = cfg.configDir != null || cfg.rootpwFile != null || cfg.rootpw != null;
-        message = "services.openldap: Unless configDir is set, either rootpw or rootpwFile must be set";
-      }
-    ];
+    warnings = let
+      deprecations = [
+        { old = "logLevel"; new = "attrs.olcLogLevel"; }
+        { old = "defaultSchemas";
+          new = "children.\"cn=schema\".includes";
+          newValue = "[\n    ${lib.concatStringsSep "\n    " [
+            "\${pkgs.openldap}/etc/schema/core.ldif"
+            "\${pkgs.openldap}/etc/schema/cosine.ldif"
+            "\${pkgs.openldap}/etc/schema/inetorgperson.ldif"
+            "\${pkgs.openldap}/etc/schema/nis.ldif"
+          ]}\n  ]"; }
+        { old = "database"; new = "children.\"cn={1}${cfg.database}\""; newValue = "{ }"; }
+        { old = "suffix"; new = "children.\"cn={1}${cfg.database}\".attrs.olcSuffix"; }
+        { old = "dataDir"; new = "children.\"cn={1}${cfg.database}\".attrs.olcDbDirectory"; }
+        { old = "rootdn"; new = "children.\"cn={1}${cfg.database}\".attrs.olcRootDN"; }
+        { old = "rootpw"; new = "children.\"cn={1}${cfg.database}\".attrs.olcRootPW"; }
+        { old = "rootpwFile";
+          new = "children.\"cn={1}${cfg.database}\".attrs.olcRootPW";
+          newValue = "{ path = \"${cfg.rootpwFile}\"; }";
+          note = "The file should contain only the password (without \"rootpw \" as before)"; }
+      ];
+    in (optional (cfg.extraConfig != "" || cfg.extraDatabaseConfig != "") ''
+      The options `extraConfig` and `extraDatabaseConfig` of `services.openldap`
+      are deprecated. This is due to the deprecation of `slapd.conf`
+      upstream. Please migrate to `services.openldap.settings`.
```

Comment:
How about using `mkRemovedOptionModule` instead?

Reply:
Sure, I wanted to support this for one release while displaying the warning, but I'm happy to remove them!